tangelo/toolboxes/operators/trim_trivial_qubits.py

```diff
@@ -0,0 +1,171 @@
+# Copyright 2023 Good Chemistry Company.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numpy as np
+
+from tangelo.toolboxes.operators import QubitOperator, count_qubits
+from tangelo.linq import Circuit
+from tangelo.linq.helpers.circuits import pauli_string_to_of, pauli_of_to_string
+
+
+def trim_trivial_operator(qu_op, trim_states, n_qubits=None, reindex=True):
+    """
+    Calculate expectation values of a QubitOperator acting on qubits in a
+    trivial |0> or |1> state. Return a trimmed QubitOperator with updated coefficients
+
+    Args:
+        qu_op (QubitOperator): Operator to trim
+        trim_states (dict): Dictionary mapping qubit indices to states to trim, e.g. {1: 0, 3: 1}
+        n_qubits (int): Optional, number of qubits in full system
+        reindex (bool): Optional, if True, remaining qubits will be reindexed
+    Returns:
+        QubitOperator : trimmed QubitOperator with updated coefficients
+    """
+
+    qu_op_trim = QubitOperator()
+    n_qubits = count_qubits(qu_op) if n_qubits is None else n_qubits
+
+    # Calculate expectation values of trivial qubits, update coefficients
+    for op, coeff in qu_op.terms.items():
+        term = pauli_of_to_string(op, n_qubits)
+        c = np.ones(len(trim_states))
+        new_term = term
+        for i, qubit in enumerate(trim_states.keys()):
+            if term[qubit] in {'X', 'Y'}:
+                c[i] = 0
+                break
+            elif (term[qubit], trim_states[qubit]) == ('Z', 1):
+                c[i] = -1
+
+            new_term = new_term[:qubit - i] + new_term[qubit - i + 1:] if reindex else new_term[:qubit] + 'I' + new_term[qubit + 1:]
+
+        if 0 in c:
+            continue
+
+        qu_op_trim += np.prod(c) * coeff * QubitOperator(pauli_string_to_of(new_term))
+    return qu_op_trim
+
+
+def is_bitflip_gate(gate, atol=1e-5):
+    """
+    Check if a gate is a bitflip gate.
+
+    A gate is a bitflip gate if it satisfies one of the following conditions:
+    1. The gate name is either X, Y.
+    2. The gate name is RX or RY, and has a parameter that is an odd multiple of pi.
+
+    Args:
+        gate (Gate): The gate to check.
+        atol (float): The absolute tolerance for gate parameter
```

Comment:
Nagging you until the very last change ;)

"Optional, the absolute tolerance for..."